test/com/google/javascript/jscomp/RewriteAsyncFunctionsTest.java

```diff
@@ -142,6 +142,74 @@ public void testInnerSuperReference() {
             "}"));
   }
 
+  @Test
+  public void testInnerSuperCallEs2015Out() {
+    setLanguageOut(LanguageMode.ECMASCRIPT_2015);
+    test(
+        lines(
+            "class A {",
+            "  m() {",
+            "    return this;",
+            "  }",
+            "}",
+            "class X extends A {",
+            "  async m() {",
+            "    return super.m();",
+            "  }",
+            "}"),
+        lines(
+            "class A {",
+            "  m() {",
+            "    return this;",
+            "  }",
+            "}",
+            "class X extends A {",
+            "  m() {",
+            "    const $jscomp$async$this = this;",
+            "    const $jscomp$async$super$get$m =",
+            "        () => Object.getPrototypeOf(this.constructor).prototype.m;",
```

Comment:
Couldn't we just get the prototype directly from `this`?
```javascript
() => Object.getPrototypeOf(this).m;
```

Reply:
Unfortunately not. See the comments in the following snippet:

```js
class Base {
  static foo() { console.log('base foo'); }
  bar() { console.log('base bar'); }
}

class Child extends Base {
  static foo() {
    console.log(Object.getPrototypeOf(this), Object.getPrototypeOf(this).foo);
    Object.getPrototypeOf(this).foo.call(this);
  }

  bar() {
    console.log(Object.getPrototypeOf(this), Object.getPrototypeOf(this).bar);
    Object.getPrototypeOf(this).bar(); // logs out "base bar" appropriately
    Object.getPrototypeOf(this).bar.call(this); // infinitely loops and locks up the Chrome tab!
  }
}
```

Reply:
Ah, I see my mistake.
I meant to say `Object.getPrototypeOf(Object.getPrototypeOf(this))`

I wonder which is the safer choice in the face of code that mucks around with prototypes?
Is it more likely that `this.constructor` isn't pointing at the right thing or that an additional object has been inserted into the prototype chain?

Reply:
Discussed this with @concavelenz 
We agree that it seems more correct and slightly safer to call `Object.getPrototypeOf()` twice to get up to the superclass prototype rather than relying on `this.constructor`.

Would you be willing to make that change?

Reply:
Of course - done.